### ISSUE TITLE
Docs for differenceInDays should mention or show an example with negative results

### ISSUE THREAD

**Issue by hgezim** [date-fns/date-fns#2888]
It wasn't clear that there would be any negative results when using `differenceInDays`.

**Comment by grantralls** [comment]
I ran into this same issue with the docs, I'm opening a PR right now that adds a bit more clarification.

**Comment by pip8786** [comment]
Somewhat related to #2179 and it's associated PR (which I don't think solves the issue. I don't think you can have a PR that only does it for one of those functions as I'm sure they all work the same way.

**Comment by grantralls** [comment]
@pip8786 You're right. Maybe they'll see this as an incremental improvement made to be added to other documentation over time 🤷. I'll take a look tomorrow to see how feasible it is to do them in one PR. At least all of the "differenceIn" functions.

**Comment by shawnlee-info** [comment]
I ran into this issue by simply thinking leftDate parameter is the Start Date, and the rightDate is End Date, so the result should be naturally positive "how many days in between", but actually it's opposite, I highly recommend WRITE A FUNCTION NAME WITH COMMON SENSE, how do you usually book a hotel? when you choose leftDate, rightDate, how many days would you stay? negative?

**Comment by shawnlee-info** [comment]
The serious problem isn't the function name, it's the parameter name in the doc: 

- DateLeft (which actually means the End Date)
- DateRight (which actually means the Start Date)
What brain comes out such a naming? and look at your doc example:

`const result = differenceInDays(
  new Date(2012, 6, 2, 0, 0),
  new Date(2011, 6, 2, 23, 0)
)
//=> 365`

So 2012-06-02 is the "DateLeft" accordingly, 2011-06-02 is the "DateRight", how come this not making you feel weird?
